sources/packages/backend/apps/api/src/route-controllers/student-appeal/student-appeal.students.controller.ts

```diff
@@ -145,6 +145,43 @@ export class StudentAppealStudentsController extends BaseController {
         } as StudentAppealRequestModel),
     );
 
+    // Obtain the data for validDependent and isavalidDependant APIs for each dependant
+    // for student dependents appeal if it has dependents
+    for (let appealRequest of appealRequests) {
```

Comment:
I have 2 questions here.

1 - Do we really need the hidden variables `validDependent` `isavalidDependant`, they look like obsolete to me.

2 - If they are required, we can use form io to do the logic as per student application.
To populate the hidden variables on submission, please do it from form.io by enabling this option.

![image](https://github.com/bcgov/SIMS/assets/54600590/740f3a74-1b38-4142-9fa8-f67adaf4be26)

Reply:
It may be obsolete but it would require a further looking.
The form.io logic should never be replicated in the server and the `Calculated on server` must be used for similar cases.

Reply:
The reason I moved them to the server is because I would need to change multiple DTO files and send data from server to the form.io then back again, but the current approach is do the calculation in a centralized spot for calculation whose data come be obtained directly from the server. I see now that we should do the calculation on the form.io not on the server. I will a dd the calculation for `validDependent` and `isavalidDependant` on the form.io.

Reply:
To be clear, as much as possible the custom validations in form.io definition should be as minimal as possible, but once they are there we do not need to have them replicated on the server-side and we must ensure the `Calculate on server` config as true.

Reply:
As @dheepak-aot mentioned, these values calculations can be moved to formio server side calculation and retrieved accordingly.

Reply:
@dheepak-aot From my understanding, we will still need the hidden variables `validDependent` and `isavalidDependant` and since we do the calculations on the formio and then send it to the api everywhere else in the application, I believe we can follow the same approach here as well.

Reply:
After having the discussion with @andrewsignori-aot and @dheepak-aot, we found that `validDependent` and `isavalidDependant` are not required anywhere else, and similar validations are being done in Camunda. Searching validDependent or isavalidDependant in VS Code shows that they appear only in "sfaa" files. For now it's safe to remove them from this PR, and I will create a story to remove them in the application form. It could be a good first issue. FYI @guru-aot @shaangill025 @andrepestana-aot 🙂